alpa/collective/collective_group/xla_nccl_collective_group.py

```diff
@@ -0,0 +1,393 @@
+"""NCCL-based collective operations with apis from xla extension."""
+import logging
+
+import ray
+from jax._src.lib import xla_extension as xe
+
+from alpa.collective.collective_group import xla_nccl_util
+from alpa.collective.collective_group.nccl_collective_group import Rendezvous
+from alpa.collective.collective_group.base_collective_group import BaseGroup
+from alpa.collective.const import get_store_name, ENV
+from alpa.collective.types import (Backend, BroadcastOptions, AllReduceOptions,
+                                   BarrierOptions, ReduceOptions,
+                                   AllGatherOptions, ReduceScatterOptions,
+                                   SendOptions, RecvOptions)
+
+logger = logging.getLogger(__name__)
+
+
+class XLANCCLGroup(BaseGroup):
+    """NCCL-based collective operations with apis from xla extension."""
+
+    def __init__(self, world_size, rank, group_name):
+        """Init an NCCL collective group."""
+        super().__init__(world_size, rank, group_name)
+
+        # communicator and stream cache.
+        # TODO (Hao): we need a lock here...
```

Comment:
Instead of copying code, maybe move shared code into the base class?

Reply:
It's for current transition period until cupy nccl will be fully deleted later. Please allow me to keep the copy code for the convinence of future modification.